2021-09-09 21:32:40,942: Unhandled exception in listener callback
Traceback (most recent call last):
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 211, in inner
    return f(self, *args, **kwargs)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/xorg.py", line 471, in _handler
    self._handle(self._display_stop, event)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/mouse/_xorg.py", line 148, in _handle
    self.on_click(px, py, self._button(event.detail), True)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 127, in inner
    if f(*args) is False:
TypeError: on_click() takes 4 positional arguments but 5 were given
2021-09-09 21:33:01,915: Unhandled exception in listener callback
Traceback (most recent call last):
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 211, in inner
    return f(self, *args, **kwargs)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/xorg.py", line 471, in _handler
    self._handle(self._display_stop, event)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/mouse/_xorg.py", line 148, in _handle
    self.on_click(px, py, self._button(event.detail), True)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 127, in inner
    if f(*args) is False:
TypeError: on_click() takes 4 positional arguments but 5 were given
2021-09-22 22:20:16,614: Unhandled exception in listener callback
Traceback (most recent call last):
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 211, in inner
    return f(self, *args, **kwargs)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/xorg.py", line 471, in _handler
    self._handle(self._display_stop, event)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/mouse/_xorg.py", line 148, in _handle
    self.on_click(px, py, self._button(event.detail), True)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 127, in inner
    if f(*args) is False:
  File "/home/leo/Documents/Projects/macros/inputHandler.py", line 21, in on_click
    self.macrosHandler.buttonPress(button.name)
  File "/home/leo/Documents/Projects/macros/macrosHandler.py", line 80, in buttonPress
    if buttonName in self.buttons.keys():
AttributeError: 'macrosHandler' object has no attribute 'buttons'
2021-09-22 22:20:34,337: Unhandled exception in listener callback
Traceback (most recent call last):
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 211, in inner
    return f(self, *args, **kwargs)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/xorg.py", line 471, in _handler
    self._handle(self._display_stop, event)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/mouse/_xorg.py", line 148, in _handle
    self.on_click(px, py, self._button(event.detail), True)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 127, in inner
    if f(*args) is False:
  File "/home/leo/Documents/Projects/macros/inputHandler.py", line 21, in on_click
    self.macrosHandler.buttonPress(button.name)
  File "/home/leo/Documents/Projects/macros/macrosHandler.py", line 81, in buttonPress
    threadsHandler.runPressAction(self.actionsDict((buttonName)).press)
TypeError: 'dict' object is not callable
2021-09-22 22:23:10,830: Unhandled exception in listener callback
Traceback (most recent call last):
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 211, in inner
    return f(self, *args, **kwargs)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/xorg.py", line 471, in _handler
    self._handle(self._display_stop, event)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/mouse/_xorg.py", line 148, in _handle
    self.on_click(px, py, self._button(event.detail), True)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 127, in inner
    if f(*args) is False:
  File "/home/leo/Documents/Projects/macros/inputHandler.py", line 21, in on_click
    self.macrosHandler.buttonPress(button.name)
  File "/home/leo/Documents/Projects/macros/macrosHandler.py", line 81, in buttonPress
    threadsHandler.runPressAction(self.actionsDict([buttonName]).press)
TypeError: 'dict' object is not callable
2021-09-22 22:23:41,244: Unhandled exception in listener callback
Traceback (most recent call last):
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 211, in inner
    return f(self, *args, **kwargs)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/xorg.py", line 471, in _handler
    self._handle(self._display_stop, event)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/mouse/_xorg.py", line 148, in _handle
    self.on_click(px, py, self._button(event.detail), True)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 127, in inner
    if f(*args) is False:
  File "/home/leo/Documents/Projects/macros/inputHandler.py", line 21, in on_click
    self.macrosHandler.buttonPress(button.name)
  File "/home/leo/Documents/Projects/macros/macrosHandler.py", line 81, in buttonPress
    threadsHandler.runPressAction(self.actionsDict[buttonName].press)
TypeError: runPressAction() missing 1 required positional argument: 'actionFunc'
2021-09-22 22:25:09,256: Unhandled exception in listener callback
Traceback (most recent call last):
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 211, in inner
    return f(self, *args, **kwargs)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/xorg.py", line 471, in _handler
    self._handle(self._display_stop, event)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/mouse/_xorg.py", line 148, in _handle
    self.on_click(px, py, self._button(event.detail), True)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 127, in inner
    if f(*args) is False:
  File "/home/leo/Documents/Projects/macros/inputHandler.py", line 21, in on_click
    self.macrosHandler.buttonPress(button.name)
  File "/home/leo/Documents/Projects/macros/macrosHandler.py", line 81, in buttonPress
    threadsHandler.runPressAction(self.actionsDict[buttonName].press)
TypeError: runPressAction() missing 1 required positional argument: 'actionFunc'
2021-09-22 22:25:58,034: Unhandled exception in listener callback
Traceback (most recent call last):
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 211, in inner
    return f(self, *args, **kwargs)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/xorg.py", line 471, in _handler
    self._handle(self._display_stop, event)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/mouse/_xorg.py", line 148, in _handle
    self.on_click(px, py, self._button(event.detail), True)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 127, in inner
    if f(*args) is False:
  File "/home/leo/Documents/Projects/macros/inputHandler.py", line 21, in on_click
    self.macrosHandler.buttonPress(button.name)
  File "/home/leo/Documents/Projects/macros/macrosHandler.py", line 82, in buttonPress
    threadsHandler.runPressAction(self.actionsDict[buttonName].press)
TypeError: runPressAction() missing 1 required positional argument: 'actionFunc'
2021-09-22 22:26:11,775: Unhandled exception in listener callback
Traceback (most recent call last):
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 211, in inner
    return f(self, *args, **kwargs)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/xorg.py", line 471, in _handler
    self._handle(self._display_stop, event)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/mouse/_xorg.py", line 148, in _handle
    self.on_click(px, py, self._button(event.detail), True)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 127, in inner
    if f(*args) is False:
  File "/home/leo/Documents/Projects/macros/inputHandler.py", line 21, in on_click
    self.macrosHandler.buttonPress(button.name)
  File "/home/leo/Documents/Projects/macros/macrosHandler.py", line 82, in buttonPress
    threadsHandler.runPressAction(self.actionsDict[buttonName].press)
TypeError: runPressAction() missing 1 required positional argument: 'actionFunc'
2021-09-22 22:26:30,552: Unhandled exception in listener callback
Traceback (most recent call last):
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 211, in inner
    return f(self, *args, **kwargs)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/xorg.py", line 471, in _handler
    self._handle(self._display_stop, event)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/mouse/_xorg.py", line 148, in _handle
    self.on_click(px, py, self._button(event.detail), True)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 127, in inner
    if f(*args) is False:
  File "/home/leo/Documents/Projects/macros/inputHandler.py", line 21, in on_click
    self.macrosHandler.buttonPress(button.name)
  File "/home/leo/Documents/Projects/macros/macrosHandler.py", line 82, in buttonPress
    self.threadsHandler.runPressAction(self.actionsDict[buttonName].press)
AttributeError: 'macrosHandler' object has no attribute 'threadsHandler'
2021-09-22 22:26:47,674: Unhandled exception in listener callback
Traceback (most recent call last):
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 211, in inner
    return f(self, *args, **kwargs)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/xorg.py", line 471, in _handler
    self._handle(self._display_stop, event)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/mouse/_xorg.py", line 153, in _handle
    self.on_click(px, py, self._button(event.detail), False)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 127, in inner
    if f(*args) is False:
  File "/home/leo/Documents/Projects/macros/inputHandler.py", line 25, in on_click
    self.macrosHandler.buttonRelease(button.name)
  File "/home/leo/Documents/Projects/macros/macrosHandler.py", line 88, in buttonRelease
    self.threadsHandler.runReleaseAction(self.actionsDict[buttonName].release)
TypeError: runReleaseAction() missing 1 required positional argument: 'buttonName'
2021-09-22 22:26:59,122: Unhandled exception in listener callback
Traceback (most recent call last):
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 211, in inner
    return f(self, *args, **kwargs)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/xorg.py", line 471, in _handler
    self._handle(self._display_stop, event)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/mouse/_xorg.py", line 153, in _handle
    self.on_click(px, py, self._button(event.detail), False)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 127, in inner
    if f(*args) is False:
  File "/home/leo/Documents/Projects/macros/inputHandler.py", line 25, in on_click
    self.macrosHandler.buttonRelease(button.name)
  File "/home/leo/Documents/Projects/macros/macrosHandler.py", line 88, in buttonRelease
    self.threadsHandler.runReleaseAction(self.actionsDict[buttonName].release, buttonName)
  File "/home/leo/Documents/Projects/macros/threadsHandler.py", line 21, in runReleaseAction
    self.holdThreads[buttonName].stop()
AttributeError: 'Thread' object has no attribute 'stop'
2021-09-22 23:00:26,922: Unhandled exception in listener callback
Traceback (most recent call last):
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 211, in inner
    return f(self, *args, **kwargs)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/xorg.py", line 471, in _handler
    self._handle(self._display_stop, event)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/mouse/_xorg.py", line 153, in _handle
    self.on_click(px, py, self._button(event.detail), False)
  File "/home/leo/.local/lib/python3.8/site-packages/pynput/_util/__init__.py", line 127, in inner
    if f(*args) is False:
  File "/home/leo/Documents/Projects/macros/inputHandler.py", line 25, in on_click
    self.macrosHandler.buttonRelease(button.name)
  File "/home/leo/Documents/Projects/macros/macrosHandler.py", line 88, in buttonRelease
    self.threadsHandler.runReleaseAction(self.actionsDict[buttonName].release, buttonName)
  File "/home/leo/Documents/Projects/macros/threadsHandler.py", line 21, in runReleaseAction
    self.holdThreads[buttonName].stop()
AttributeError: 'Thread' object has no attribute 'stop'
